### PR TITLE
Implement /{db}/_purge operation

### PR DESCRIPTION
**Pull request by ajres**
fixes issue #1219

### CODE REVIEW

File: src/github.com/couchbase/sync_gateway/rest/admin_api_test.go [hunk 1]
```diff
@@ -908,3 +908,71 @@ func (rt *restTester) createSession(t *testing.T, username string) string {
...
+
+func TestPurgeWithNonArrayRevisionList(t *testing.T) {
+	var rt restTester
+	response := rt.sendAdminRequest("POST", "/db/_purge", `{"foo":"list"}`)
+	assertStatus(t, response, 200)
+}
+
+func TestPurgeWithEmptyRevisionList(t *testing.T) {
+	var rt restTester
+	response := rt.sendAdminRequest("POST", "/db/_purge", `{"foo":[]}`)
```
Comment (adamcfraser) [inline src/github.com/couchbase/sync_gateway/rest/admin_api_test.go:931]: Is there more we can be validating for some of these tests (e.g. the number of docs purged, etc)?

File: src/github.com/couchbase/sync_gateway/rest/admin_api.go [hunk 2]
```diff
@@ -374,3 +374,68 @@ func (h *handler) handleIndexAllChannels() error {
...
+				s := fmt.Sprintf("\"%v\" : [\"*\"]\n", key)
+				h.response.Write([]byte(s))
+
+			} else {
+				base.LogTo("CRUD", "Failed to purge document %v, err = %v", key, err)
+				continue //skip this entry its not valid
+			}
+
+		} else {
+			base.LogTo("CRUD","Revision list for doc ID %v, is not an array, ", key)
```
Comment (adamcfraser) [inline src/github.com/couchbase/sync_gateway/rest/admin_api.go:432]: If this is a badly formatted request, should we return an error here instead?  I guess the problem is that we might have already purged previous entries?

Comment (ajres) [inline src/github.com/couchbase/sync_gateway/rest/admin_api.go:432]: Yes, the design ignores failure on individual doc ID's as previous docID's will already have been purged. The only indication that an entry failed is its absence from the response payload.

File: src/github.com/couchbase/sync_gateway/rest/admin_api_test.go (hunk 1 above)
Comment (ajres) [inline src/github.com/couchbase/sync_gateway/rest/admin_api_test.go:931]: The later success tests validate the response, I can do the same here to validate the response only contains the boilerplate structure

Comment (ajres) [inline src/github.com/couchbase/sync_gateway/rest/admin_api_test.go:931]: Additional validation on the responses has been added

**Pull request merged**